### PR TITLE
Optimized while loop, implamented QbCore Functions

### PR DESCRIPTION
**Pull request by Rav3n95**

### CODE REVIEW

**Comment by BerkieBb** [comment]
The purpose of having those local functions is optimization, using local function is faster than using that of an object

File: client.lua [hunk 1]
```diff
@@ -99,17 +49,21 @@ RegisterKeyMapping('scoreboard', 'Open Scoreboard', 'keyboard', Config.OpenKey)
 -- Threads
 
 CreateThread(function()
+    local loop = 1000
```
Comment (BerkieBb) [inline client.lua:52]: This can be declared inside the loop instead

File: client.lua [hunk 2]
```diff
@@ -99,17 +49,21 @@ RegisterKeyMapping('scoreboard', 'Open Scoreboard', 'keyboard', Config.OpenKey)
...
                 local PlayerId = GetPlayerServerId(player)
                 local PlayerPed = GetPlayerPed(player)
                 local PlayerCoords = GetEntityCoords(PlayerPed)
                 if Config.ShowIDforALL or PlayerOptin[PlayerId].permission then
-                    DrawText3D(PlayerCoords.x, PlayerCoords.y, PlayerCoords.z + 1.0, '['..PlayerId..']')
+                    QBCore.Functions.DrawText3D(PlayerCoords.x, PlayerCoords.y, PlayerCoords.z + 1.0, '['..PlayerId..']')
                 end
             end
+            loop = 0
+        else
```
Comment (BerkieBb) [inline client.lua:64]: This else is unnecessary if you just declare the loop variable inside the loop

File: client.lua [hunk 3]
```diff
@@ -99,17 +49,21 @@ RegisterKeyMapping('scoreboard', 'Open Scoreboard', 'keyboard', Config.OpenKey)
...
+            for _, player in pairs(QBCore.Functions.GetPlayersFromCoords(GetEntityCoords(PlayerPedId()), 10.0)) do
                 local PlayerId = GetPlayerServerId(player)
                 local PlayerPed = GetPlayerPed(player)
                 local PlayerCoords = GetEntityCoords(PlayerPed)
                 if Config.ShowIDforALL or PlayerOptin[PlayerId].permission then
-                    DrawText3D(PlayerCoords.x, PlayerCoords.y, PlayerCoords.z + 1.0, '['..PlayerId..']')
+                    QBCore.Functions.DrawText3D(PlayerCoords.x, PlayerCoords.y, PlayerCoords.z + 1.0, '['..PlayerId..']')
                 end
             end
+            loop = 0
```
Comment (BerkieBb) [inline client.lua:63]: Place this inside the if statement instead of outside it so it doesn't wait 0 for nothing

**Review by BerkieBb — CHANGES_REQUESTED** [review]

**Comment by Rav3n95** [comment]
Let me fix these tomorrow.

**Comment by Rav3n95** [comment]
> The purpose of having those local functions is optimization, using local function is faster than using that of an object

While loops are fixed, should i just revert those local function then?

**Comment by BerkieBb** [comment]
Yes please

**Comment by Rav3n95** [comment]
> Yes please

Sure, give me a minute. I just tougth its impelented to core for reasaon.

**Comment by BerkieBb** [comment]
> > Yes please
> 
> Sure, give me a minute. I just tougth its impelented to core for reasaon.

Those were there before we knew local functions were better to use, so now we keep them for backwards compatibility

**Comment by Rav3n95** [comment]
> > > Yes please
> > 
> > 
> > Sure, give me a minute. I just tougth its impelented to core for reasaon.
> 
> Those were there before we knew local functions were better to use, so now we keep them for backwards compatibility

Ohh i understand now, you always help a lot. Thank you.

**Review by BerkieBb — APPROVED** [review]

**Pull request merged**